### PR TITLE
Add predicted event spend support

### PR DESCRIPTION
**Pull request by corke2013**
MR to add PES functionality to the events endpoint.

### CODE REVIEW

**Review by CockyAmoeba — APPROVED** [review]
Just add explicit unit test for PES query param nested parsing

File: predicthq/endpoints/decorators.py [hunk 1]
```diff
@@ -16,16 +16,25 @@ def _to_url_params(data, glue=".", separator=","):
         elif isinstance(value, list):
             params[key] = separator.join(map(str, value))
         elif isinstance(value, dict):
-            for subkey, subvalue in value.items():
-                if isinstance(subvalue, list):
-                    params[glue.join((key, subkey))] = separator.join(map(str, subvalue))
-                else:
-                    params[glue.join((key, subkey))] = subvalue
+            params.update(**_flatten_dict(value, glue, separator, pk=key))
```
Comment (medhatphq) [inline predicthq/endpoints/decorators.py:19]: Why do you use argument unpacking here but not on line 29? should we stick to the same call type for consistency?

File: predicthq/endpoints/decorators.py [hunk 2]
```diff
@@ -16,16 +16,25 @@ def _to_url_params(data, glue=".", separator=","):
...
         else:
             params[key] = value
     return params
 
 
+def _flatten_dict(d, glue, separator, pk=""):
```
Comment (medhatphq) [inline predicthq/endpoints/decorators.py:25]: let's use more explicit var names for easier readability, e.g. does pk mean primary key? or parent key?

**Review by medhatphq — APPROVED** [review]

File: predicthq/endpoints/decorators.py (hunk 1 above)
Comment (corke2013) [reply predicthq/endpoints/decorators.py:19]: Good catch, updating now.

File: predicthq/endpoints/decorators.py (hunk 2 above)
Comment (corke2013) [reply predicthq/endpoints/decorators.py:25]: Done

**Pull request merged**